lib/process.gi

```diff
@@ -261,3 +261,77 @@ InstallGlobalFunction( Exec, function( arg )
     Process( dir, shell, InputTextUser(), OutputTextUser(), [ cs, cmd ] );
 
 end );
+
+# TODO: document this, come up with a better name, write some tests...
+BindGlobal( "Exec2", function( arg )
+    local args, result, a, input, output, dir, cmd;
+
+    args := [];
+    result := rec();
+
+    # parse the inputs
+    for a in arg do
+        if IsDirectory(a) then
+            if IsBound(dir) then
+                Error("must specify at most one working directory");
+            fi;
+            dir := a;
+        elif IsInputStream(a) then
```

Comment:
Right now input and output stream can come at any position and in any order. Perhaps I should make this code stricter, and require that they come in a specific place (canonical places: 1. as the first two arguments; 2. after the command name; 3. as the last two arguments)

Reply:
While it would be a bigger change, I wonder if the input should be a record, with keys like `input :=`, `output := `.. this would mean in future it would be easier to do things like add stderr support, and make it clear which output stream is stdout and which is stderr, things like that.

Reply:
I like that idea. It could then also be made backwards compatible: if there is a single argument which is a record, then use the new code; otherwise call the old code.

Reply:
What I don't like about it is that for simple uses, it is much more verbose. Compare: e.g.:

old code:
```gap
Exec( Concatenation("start ", winfilename ) );
```
this PR:
```gap
Exec2( "start", winfilename );
```
with a record (record entries subject to change, of course):
```gap
Exec(rec(cmd:= "start", args := [ winfilename ] );
```
A compromise would be to only use a `rec` for the "extra" arguments like input, output, etc., like so:

```gap
Exec2( "start", winfilename, rec( input := BLAH, output := BLEH ) );
```
but then of course I can't use this to differentiate old and new callers.